alinka/widget/components.py

```diff
@@ -221,20 +222,48 @@ def clear(self) -> None:
 
 
 class LabeledDateComponent(ValidationMixin, QFrame):
-    def __init__(self, text, parent):
+    def __init__(self, text, parent, required: bool = False):
         super().__init__(parent)
+        self.required = required
         layout = QVBoxLayout(self)
         layout.setContentsMargins(0, 0, 0, 0)
         label = QLabel(text=text, parent=self)
         self.date_input = QDateEdit(self)
         self.date_input.setCalendarPopup(True)
-        self.date_input.dateTime()
         layout.addWidget(label)
         layout.addWidget(self.date_input)
 
     def clear(self) -> None:
         self.date_input.clear()
 
+    def display_validation_result(self, validation_result: bool) -> None:
+        if validation_result:
+            self.toggle_highlight("lightgreen")
+        else:
+            self.toggle_highlight("mistyrose")
+
+    def toggle_highlight(self, color: str | None) -> None:
+        if color:
+            self.date_input.setStyleSheet(f"background-color: {color};")
+        else:
+            self.date_input.setStyleSheet("")
+
+    @property
+    def is_valid(self) -> bool:
+        if self.required and not self.date_input.date():
+            return False
+        return True
```

Comment:
I know that it can be written shorter:
```
@property
def is_valid(self) -> bool:
    return not (self.required and not self.date_input.date())
```
But both way have the same performance, but current one is less cognitive complex, and easier to read.